differential/customfield/DifferentialBugzillaBugIDField.php

```diff
@@ -64,14 +61,42 @@ public function readValueFromRequest(AphrontRequest $request) {
   }
 
   public function renderEditControl(array $handles) {
-    // TODO add validation
     return id(new AphrontFormTextControl())
       ->setLabel($this->getFieldName())
       ->setCaption(
         pht('Example: %s', phutil_tag('tt', array(), '2345')))
       ->setName($this->getFieldKey())
-      ->setValue($this->getValue(), '')
-      ->setError($this->error);
+      ->setValue($this->getValue(), '');
+  }
+
+  public function validateApplicationTransactions(
+    PhabricatorApplicationTransactionEditor $editor,
+    $type, array $xactions) {
+
+    $errors = parent::validateApplicationTransactions($editor, $type, $xactions);
+
+    foreach ($xactions as $xaction) {
+      // Get the transactor's ExternalAccount->accountID using the author's phid
+      $xaction_author_phid = $xaction->getAuthorPHID();
```

Comment:
We should do a check here to see if the user is a Phabricator admin.  If the user is an admin, we should skip the validation checks, since they may not have a BMO ID and thus the bug permission validation would fail.